wss.js

```diff
@@ -1,18 +1,50 @@
 const WebSocket = require('ws');
+let { BruinBot } = require('./models/bruinbot.model');
+let { coordDistanceM } = require('./util/utils');
+let { VICINITY } = require('./constants');
 
 const wss = new WebSocket.Server({ noServer: true });
 
+const updateLocationHandler = async function (botId, lat, lon) {
+	try {
+		let bot = await BruinBot.findById(botId);
+
+		if (!bot) return;
+
+		bot.location.latitude = lat;
+		bot.location.longitude = lon;
+
+		if (bot.status == 'InTransit') {
+			let pathEnd = bot.path[bot.path.length - 1];
+			if (
+				coordDistanceM(lat, lon, pathEnd.latitude, pathEnd.longitude) < VICINITY
+			) {
+				bot.path = [];
+				bot.status = 'Idle';
+			}
+		}
+
+		await bot.save();
+		console.log(`Successfully updated location of bot ${botId}`);
+	} catch (err) {
+		console.log('Error: ' + err);
+	}
+};
+
 const messageHandler = (msg) => {
-	console.log(`received: ${msg}`);
+	const msgSplit = msg.split(' ');
+	console.log(msg);
+	if (msgSplit[0] == 'LOCATION' && msgSplit.length == 4) {
```

Comment:
Really small thing, but I think Max is going with lowercase for the message command, so might want to keep that consistent?

Reply:
Sounds good! I'll update those commands.